src/main/kotlin/com/sourcegraph/cody/edit/widget/LensSpinner.kt

```diff
@@ -0,0 +1,61 @@
+package com.sourcegraph.cody.edit.widget
+
+import com.intellij.openapi.application.ApplicationManager
+import java.awt.FontMetrics
+import java.awt.Graphics2D
+import java.awt.geom.AffineTransform
+import javax.swing.Icon
+import javax.swing.Timer
+
+class LensSpinner(group: LensWidgetGroup, private val icon: Icon) : LensWidget(group) {
```

Comment:
I'm not sure if I get purpose of this class correctly, but IntelliJ supports [animated icons](https://plugins.jetbrains.com/docs/intellij/icons.html#animated-icons) for spinners (see: `AnimatedIcon.Default`).
If your use case requires something completely custom could elaborate on why?

Reply:
I've taken a look and I would love to replace this code with the AnimatedIcon. But I judge it to be more than an hour's work, so I've filed [issue 1142](https://github.com/sourcegraph/jetbrains/issues/1142) to track it.